tests/src/tracing/tracevalidation/jittingstarted/JittingStarted.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using Tracing.Tests.Common;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+
+namespace Tracing.Tests
+{
+    public static class TraceValidationJittingStarted
+    {
+        public static int Main(string[] args)
+        {
+            bool pass = true;
+            bool keepOutput = false;
+
+            // Use the first arg as an output filename if there is one.
+            string outputFilename = null;
+            if (args.Length >= 1)
+            {
+                outputFilename = args[0];
+                keepOutput = true;
+            }
+            else
+            {
+                outputFilename = System.IO.Path.GetTempPath() + Guid.NewGuid().ToString() + ".netperf";
+            }
+
+            try
+            {
+                Console.WriteLine("\tStart: Enable tracing.");
+                TraceControl.EnableDefault(outputFilename);
+                Console.WriteLine("\tEnd: Enable tracing.\n");
+
+                Console.WriteLine("\tStart: Generate some events.");
+                CompiledMethod();
```

Comment:
In [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet/blob/7572f4dd18e80e8f5c24884a771a9fdef022f171/src/BenchmarkDotNet.Core/Templates/BenchmarkProgram.txt#L29) I have once faced an issue with assembly loading. It tuned out that If method A calls method B, method B might get jitted together with A.

To make sure that the jitting happens exactly when we want it to happen I would:

1.  Create new class "NotJitted"
2. Move "CompiledMethod" there
3. Call it from here via reflection. Just to make sure there are no strong dependencies between these two types and jitting will happen when we want it to happen.

```cs
Type type = typeof(TraceValidationJittingStarted).GetTypeInfo().Assembly.GetType("Tracing.Tests.NotJitted");

type.GetMethod("CompiledMethod", BindingFlags.NonPublic | BindingFlags.Static).Invoke(null, null);
```

Reply:
I was thinking of doing something like this - but putting the method in a different assembly to trick the test harness into not running crossgen on it.  However, there are a few different ways in which tests are run and I don't want to track them all down.  So, instead I used Reflection.Emit to produce an empty dynamic method which crossgen can't see through.  Thanks!